### DISCUSSION TITLE
How to detect running under WineSkin vs Windows

### DISCUSSION THREAD

**Discussion by rosslevis** [The-Wineskin-Project/WineskinServer#505] [General]
I'm a software developer using Delphi 7 and we have an issue with a program that doesn't fully terminate when exiting, but only on Mac OS under WineSkin.  The window disappears but task remains operating.  I fixed the issue by forcing a Win32 "ExitProcess" command, but we would like to only do this when running under WineSkin and not Windows.  In the past I used
WINE := GetProcAddress(GetModuleHandle('ntdll'),'wine_get_host_version') <> nil;
But this doesn't work any longer.  Is there something else I can use?  Thank you.

**Comment by Gcenx** [comment]
Try using `__wine_dbg_output`, that’s used to detect if running under a wine environment by DXVK.

**Reply by rosslevis** [reply]
I see my issue now.  It should be 'ntdll.dll'.  Stupid mistake.  thanks anyway.
